### DISCUSSION TITLE
How to filter out null values on JSONB field

### DISCUSSION THREAD

**Discussion by benedyktdryl** [prisma/prisma#8373] [Q&A] [answered]
Hi,

I hope I am trying to find an answer in right place. I would like to ask because it's not clear for me looking at documentation and errors I receive, how I can filter by non-null values on JSONB fields.

I am using it like this:

```typescript
prisma.model.findMany({
  select: { ... },
  where: {
    NOT: {
      data: { path: ['foo'], equals: null }
    }
  }
})
```

This code is giving me errors like:

`Argument equals for where.NOT.0.data.equals must not be null. Please use undefined instead.`

Looking at [THIS](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined) link I thought it's the proper approach and switched to `undefined`. An error has gone but Prisma keeps returning data with null value on `foo` property.

Per my understanding, `undefined` isn't `null`, this value is defined in DB and explicitly have `null` value set so it makes sense Prisma is not filtering it out. However, I cannot use `null` either. Nor `null`, neither `"null"` or `"NULL"`. 

Can you tell me if I missed something? Special type for nullish value representation? Or it's rather a bug?

**Comment by ryands17** [comment, marked as answer]
@benedyktdryl 👋 

Seems like a bug to me. I would suggest opening an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) so that we can look into this 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
